test/serializers/serializer_for_test.rb

```diff
@@ -59,6 +70,18 @@ def test_serializer_custom_serializer
           serializer = ActiveModel::Serializer.serializer_for(@custom_profile)
           assert_equal ProfileSerializer, serializer
         end
+
+        def test_serializer_for_namespaced_resource
+          post = ResourceNamespace::Post.new
+          serializer = ActiveModel::Serializer.serializer_for(post)
+          assert_equal(ResourceNamespace::PostSerializer, serializer)
+        end
+
+        def test_serializer_for_nested_resource
+          comment = ResourceNamespace::Comment.new
+          serializer = ResourceNamespace::PostSerializer.serializer_for(comment)
+          assert_equal(ResourceNamespace::PostSerializer::CommentSerializer, serializer)
```

Comment:
We need to add a test to `test/serializers/associations_test.rb` to make sure the corresponding serializer gets used. This will be doable in a decent way once the test infrastructure refactor has been done.